bundles/org.openhab.binding.radiothermostat/src/main/java/org/openhab/binding/radiothermostat/internal/handler/RadioThermostatHandler.java

```diff
@@ -288,12 +288,14 @@ public void dispose() {
         connector.removeEventListener(this);
 
         // Disable Remote Temp and Message Area on shutdown
-        if (isLinked(REMOTE_TEMP)) {
-            connector.sendCommand("rem_mode", "0", REMOTE_TEMP_RESOURCE);
-        }
+        if (ThingStatus.ONLINE.equals(this.getThing().getStatus())) {
+            if (isLinked(REMOTE_TEMP)) {
+                connector.sendCommand("rem_mode", "0", REMOTE_TEMP_RESOURCE);
```

Comment:
If there is no need to do this when the thing is not online, that helps.

But the real issue is probably this blocing call. `dispose` is expected to be non-blocking and to return quickly - see https://www.openhab.org/docs/developer/bindings/#shutdown. So you should probably schedule a job for sending these commands.

Reply:
Gotcha, I have adjusted the code to run the calls within a job.

Reply:
I'm now wondering if there is a risk that `isLinked` will not work after `dispose` returns. I didn't check the implementation (yet), but in such a case, introducing local variables before scheduling the job might help.